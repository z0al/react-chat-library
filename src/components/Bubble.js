// @ts-check

// Packages
import React from 'react';

// Ours
import Text from './Text';
import Time from './Time';

/**
 * @typedef Props
 * @property {string} text
 * @property {string} date
 * @property {string} [position]
 * @property {{left?: any,right?: any}} [style]
 * @param {Props} props
 */
const Bubble = props => (
	<div
		className={`bubble ${props.position}`}
		style={props.style[props.position]}
	>
		<Text {...props} />
		<Time {...props} />
		<style jsx>{`
			div.bubble {
				display: inline-flex;
				flex-direction: column;
				padding: 0.5em 0.7em;
				margin: 0.25em 0.5em;
				max-width: 70%;
				border-radius: 1em;
			}
			div.bubble.left {
				align-self: flex-start;
				border-top-left-radius: 0px;
				background: white;
				box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.2);
				color: black;
			}
			div.bubble.right {
				align-self: flex-end;
				background: #007bef;
				border-top-right-radius: 0px;
				color: white;
			}
		`}</style>
	</div>
);

/** @type Props */
Bubble.defaultProps = {
	text: '',
	date: '',
	position: 'left',
	style: {}
};

export default Bubble;
