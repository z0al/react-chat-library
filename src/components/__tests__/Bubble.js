import React from 'react';
import renderer from 'react-test-renderer';

import Bubble from '../Bubble';

test('should match the snapshot', () => {
	const tree = renderer.create(<Bubble />).toJSON();
	expect(tree).toMatchSnapshot();
});
