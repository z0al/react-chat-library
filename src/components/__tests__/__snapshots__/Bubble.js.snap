// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="jsx-3272633064 bubble left"
>
  <span
    className="text"
    style={Object {}}
  />
  <span
    className="jsx-1805036463 left"
    style={Object {}}
  >
    Invalid Date
  </span>
</div>
`;
